data/unaligned_labeled_mask_online_dataset.py

```diff
@@ -66,65 +73,99 @@ def sanitize(self):
-                index_B = index % self.B_size
-            else:   # randomize the index for domain B to avoid fixed pairs.
-                index_B = random.randint(0, self.B_size - 1)
-            
-            B_img_path = self.B_img_paths[index_B]
-
+        if A_label_path:
             try:
-                if len(self.B_bbox_paths) > 0: # B label is optional
-                    B_label_path = self.B_bbox_paths[index_B]
+                if B_label_path is not None:
                     B_img , B_label = crop_image(B_img_path,B_label_path,mask_delta=self.opt.online_creation_mask_delta_B,crop_delta=self.opt.online_creation_crop_delta_B,mask_square=self.opt.online_creation_mask_square_B,crop_dim=self.opt.online_creation_crop_size_B,output_dim=self.opt.load_size)
                     B,B_label = self.transform(B_img,B_label)
                 else:
                     B_img = Image.open(B_img_path).convert('RGB')
                     B = self.transform_noseg(B_img)
                     B_label = []
-        
+                
             except Exception as e:
                 print("failed to read B domain image ", B_img_path, " at index_B=", index_B)
                 print(e)
                 return None
 
-            
             return {'A': A, 'B': B, 'A_paths': A_img_path, 'B_paths': B_img_path, 'A_label': A_label, 'B_label': B_label}
         else:
             return {'A': A, 'A_paths': A_img_path,'A_label': A_label}
+                
+    def __getitem__(self, index):
+        """Return a data point and its metadata information.
+
+        Parameters:
+            index (int)      -- a random integer for data indexing
+
+        Returns a dictionary that contains A, B, A_paths and B_paths
+            A (tensor)       -- an image in the input domain
+            B (tensor)       -- its corresponding image in the target domain
+            A_paths (str)    -- image paths
+            B_paths (str)    -- image paths
+            A_label (tensor) -- mask label of image A
+        """
+        A_img_path = self.A_img_paths[index % self.A_size]  # make sure index is within then range
+        A_label_path = self.A_label_paths[index % self.A_size]
 
+        if hasattr(self,'B_img_paths') :
+            if self.opt.serial_batches:   # make sure index is within then range
+                index_B = index % self.B_size
+            else:   # randomize the index for domain B to avoid fixed pairs.
+                index_B = random.randint(0, self.B_size - 1)
+            
+            B_img_path = self.B_img_paths[index_B]
+
+            if len(self.B_label_paths) > 0: # B label is optional
+                B_label_path = self.B_label_paths[index_B]
+            else:
+                B_label_path = None
+        else:
+            B_img_path=None
+                
+        return self.get_img_label(A_img_path,A_label_path,B_img_path,B_label_path)
+
+    def get_validation_set(self,size):
```

Comment:
This means that this is only available for the online mask datasets, right ? Maybe this should be applicable to all datasets ?